apis/management.cattle.io/v3/authn_types.go

```diff
@@ -175,3 +175,54 @@ type ActiveDirectoryTestAndApplyInput struct {
 	Password              string                `json:"password"`
 	Enabled               bool                  `json:"enabled,omitempty"`
 }
+
+type LdapConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	AuthConfig        `json:",inline" mapstructure:",squash"`
+
+	Servers                     []string `json:"servers,omitempty"                     norman:"type=array[string],notnullable,required"`
+	Port                        int64    `json:"port,omitempty"                        norman:"default=389,notnullable,required"`
+	TLS                         bool     `json:"tls,omitempty"                         norman:"default=false,notnullable,required"`
+	Certificate                 string   `json:"certificate,omitempty"`
+	ServiceAccountUsername      string   `json:"serviceAccountUsername,omitempty"      norman:"required"`
```

Comment:
I think we should change the name of this field to serviceAccountDistinguishedName, since it is not really a user name.